packages/application-generic/src/services/queues/queue-base.service.ts

```diff
@@ -12,14 +12,14 @@ import {
 const LOG_CONTEXT = 'QueueService';
 
 export class QueueBaseService {
-  public instance: BullMqService;
+  private instance: BullMqService;
 
   public readonly DEFAULT_ATTEMPTS = 3;
   public queue: Queue;
 
   constructor(
     public readonly topic: JobTopicNameEnum,
-    public bullMqService: BullMqService
+    private bullMqService: BullMqService
```

Comment:
This poses a major problem. Exposing the provider used for delegation means our provider becomes vulnerable to errors as it's visible externally. Additionally, tying our queue service to a specific provider's behavior will complicate switching, for instance, from Kafka to Pulsar.

Also, it's important to keep the 'queue' attribute private for now to maintain simplicity in this PR.